### PR TITLE
Feat/show principal before import pem

### PR DESCRIPTION
**Pull request by angarita-dev**
## Changelog

- Made `InfoModal` and `WalletDetailItem` a global component
- Replaced components on `WalletDetails` view
- Show principal on import pem flow

### Type of changes included:

- [x] Components
- [ ] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [x] Code Refactor

### Clubhouse Tickets Related:

- Ticket 1
- Ticket 2
- Ticket 3

### Screenshots/Gifs:

image
image

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Comment by Cancuuu** [comment]
lgtm!

File: source/Background/Keyring/index.js [hunk 1]
```diff
@@ -484,6 +485,7 @@ export const getKeyringHandler = (type, keyring) => ({
       return { error: e.message };
     }
   },
+  [HANDLER_TYPES.GET_PRINCIPAL_FROM_PEM]: async (params) => keyring.getPrincipalFromPem(params),
```
Comment (tomiir) [inline source/Background/Keyring/index.js]:
```suggestion
  [HANDLER_TYPES.GET_PRINCIPAL_FROM_PEM]: keyring.getPrincipalFromPem,
```

File: source/components/WalletDetailItem/index.jsx [hunk 2]
```diff
@@ -3,19 +3,21 @@ import PropTypes from 'prop-types';
...
-const DetailItem = ({
-  value, name, setInfoOpen, isOpen, copyButtonTestId, infoIconButtonTestId,
+const WalletDetailItem = ({
+  value, name, setInfoOpen, isOpen, copyButtonTestId, infoIconButtonTestId, className
 }) => {
   const classes = useStyles();
   const { t } = useTranslation();
   return (
-    <>
+    <div className={clsx(className?.length && className)}>
```
Comment (tomiir) [inline source/components/WalletDetailItem/index.jsx]: why not just apply it directly? an empty string does not break or change anything

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 3]
```diff
@@ -17,11 +24,26 @@ const Step2 = ({ userPemFile }) => {
...
+    fileReader.readAsText(userPemFile);
+    fileReader.onloadend = () => {
+      const content = fileReader.result;
+
+      asyncSendMessage({
+        type: HANDLER_TYPES.GET_PRINCIPAL_FROM_PEM,
+        params: { pem: content },
+      }).then(setImportedPrincipal);
+    };
+  }, [userPemFile]);
```
Comment (tomiir) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:45]: shouldn't this be wrapped under a redux action?

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx [hunk 4]
```diff
@@ -17,11 +24,26 @@ const Step2 = ({ userPemFile }) => {
   const [walletName, setWalletName] = useState(""); // add deafault wallet name, for example the next wallet not used number
   const [currentEmoji, setCurrentEmoji] = useState("😎"); // add default emoji, not used in other wallets
   const [openEmojiSelector, setOpenEmojiSelector] = useState(false);
+  const [importedPrincipal, setImportedPrincipal] = useState();
+  const [detailsModal, setDetailsModal] = useState(false);
```
Comment (tomiir) [inline source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: I'd call it `detailsModalOpen` for clarity

Comment (angarita-dev) [reply source/views/Extension/Views/ImportWallet/Steps/Step2.jsx]: Changed it 👍

File: source/views/Extension/Views/ImportWallet/Steps/Step2.jsx (hunk 3 above)
Comment (angarita-dev) [reply source/views/Extension/Views/ImportWallet/Steps/Step2.jsx:45]: mm We're not using redux for the import pem

File: source/components/WalletDetailItem/index.jsx (hunk 2 above)
Comment (angarita-dev) [reply source/components/WalletDetailItem/index.jsx]: yeah, makes sense, removed clsx

File: source/Background/Keyring/index.js (hunk 1 above)
Comment (angarita-dev) [reply source/Background/Keyring/index.js]: Done 👍

**Review by tomiir — APPROVED** [review]

**Pull request merged**